doc/direction.md

```diff
@@ -0,0 +1,52 @@
+# Minetest Direction Document
+
+
+
+## 0. Preface
+
+### 0.1. Introduction
+
+This document attempts to set out Minetest's philosophy, aims, and objectives.
+
+Please note that Minetest is developed by a community, and not everyone in the community
+holds the same views as to what Minetest should be. Therefore, defining a common direction is
+very difficult.
+
+### 0.2. Definitions
+
+* `Minetest` is the game engine.
+* `Minetest Game` is the default game which comes with the engine.
+* The `Minetest Project` refers to the combination of the above, the development team, and supporting tools.
+
+### 0.3. Other documents
+
+Please read celeron55's document: http://c55.me/blog/?p=1491
```

Comment:
This roadmap in the forum is more detailed and a continuation of that blog post https://forum.minetest.net/viewtopic.php?t=9177

Reply:
Also see celeron55's posts here https://github.com/minetest/minetest/issues/3476 for more roadmap.

Reply:
I disagree, the article contains more things relevant to direction. The thread is a roadmap, which is distinct to direction. I'll add it anyway

Reply:
Yes, sorry, it's mostly specific things to implement, but there is also some important 'direction' material in that roadmap if you read through it. So, perhaps this document can do the useful job of extracting the direction stuff from this roadmap?
```
Focus on these kinds of gameplay

    Voxel-based mechanisms that work well in multiplayer environment
        This works well together with the moddability and people have been playing with mesecons, pipes and whatnot for a long time already.
    Defending against computer-controlled enemies
        Defending instead of attacking puts much better use to the aspect of building things... at least until VAEs exist!
        PvP is generally very hard to balance without making it boring
        Many types of this require a world that is more fully in an active state than how Minetest currently works.
            Currently it is assumed that every game wants to maximize world size and player count by taking away from other things, which is not right.
            It should be possible for a mod or game to pick the tradeoff between world size, active entity count, individual entity performance and the resulting maximum player count.
```
```
Some general guidelines

    Networking: The low-level protocol implementation is bloated and not ideal. See extra information in spoiler below. Packet handling, serialization and cross-version compatibility should be made more easily reviewable and more noob-proof.
    Client UI: Keep it generic and customizable so that it works for any game and any server.
    Pull requests: If you test a pull request, always comment on it saying that you did so. The amount of testers is a very important factor in deciding whether to merge or not.
    Maintain fork-friendliness in everything including versioning of things. See http://irc.minetest.ru/minetest-dev/201 ... #i_3654822
    If you want to focus on exploration of the infinite world, you have to put way more effort into emergent generation and active systems in the world in order to actually have infinite things in there. With almost any fixed selection of static things, it's incredibly bland. For the king of this stuff, see Dwarf Fortress.
    Don't mind the lesser visuals and take full advantage of them in how they make creating unimagined things easier. There are enough games in this world that focus on visuals by taking away from everything else, or by making everything else harder to do.
    If you make game content, read this: viewtopic.php?p=139470#p139470
```

Reply:
.. and also extract the direction-related stuff from #3476